apple/swift-docc-render
# Render preview of documentation in quick navigation

Bug/issue #, if applicable: 104383742

## Summary

This updates the layout of the "Quick Navigation" feature introduced by @sofiaromorales to also show a compact preview of the documentation page for the selected result.

Example:
<img width="1436" alt="Screenshot 2023-03-01 at 1 08 08 PM" src="https://user-images.githubusercontent.com/212918/222283508-7688d504-77ad-4e1f-8fca-6a4806313a44.png">

This update currently requires a feature flag to be enabled in `theme-settings.json`, but I would like to remove that if the feedback for this new UI is generally positive.

Setting: `features.docs.quickNavigationPreview.enable` (set to `true` to opt-in)

## Testing

Generally, you should be able to just add the feature flag from above to an existing DocC project and try it out using a build of the renderer built from this branch.

Example:
1. Checkout this branch and build it with `npm run build`
2. `git clone git@github.com:apple/swift-argument-parser.git`
3. To setup swift-argument-parser repo with swift-docc-plugin and needed feature flag: `cd swift-argument-parser; curl https://gist.githubusercontent.com/mportiz08/03193ba981c168697c89c2de1fccdd6f/raw/b6b5886be9d4cc18e8764bd2d90f655a6a22c1c9/add-plugin-and-settings.diff|git apply`
4. Run the preview with `env DOCC_HTML_DIR=/path/to/swift-docc-render/dist swift package --disable-sandbox preview-documentation --target ArgumentParser`

That's just one example of course—feel free to try out previewing your own example content however you'd like.

Things to test:
* preview works for doc pages
* preview works for tutorial pages
* loading UI should show up if it takes a full second or more to load data
* "Preview unavailable" message should show if an error happens when attempting to load data for a page
* preview should generally load in a performant manner, even when scrolling results quickly
* preview should not show up without feature flag enabling it (normal quicknav experience without it)

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
